std/container/array.d

```diff
@@ -706,14 +706,32 @@ stuff)
     Array opBinary(string op, Stuff)(Stuff stuff)
     if (op == "~")
     {
-        // TODO: optimize
         Array result;
-        result ~= this[];
-        assert(result.length == length);
-        result ~= stuff[];
+
+        static if (hasLength!Stuff)
+            result.reserve(length + stuff.length);
+        else static if (is(typeof(stuff[])))
```

Comment:
`else static if (is(typeof(stuff[])) && hasLength!(typeof(stuff[])) || isSomeString!(typeof(stuff[])))`

no reason not to be careful here